lib/internal/fs/utils.js

```diff
@@ -902,8 +902,12 @@ const validateRmOptionsSync = hideStackFrames((path, options, expectDir) => {
   return options;
 });
 
-let recursiveRmdirWarned = process.noDeprecation;
+let recursiveRmdirWarned;
 function emitRecursiveRmdirWarning() {
+  if (recursiveRmdirWarned === undefined) {
+    // TODO(joyeecheung): use getOptionValue('--no-deprecation') instead.
```

Comment:
Why don't we do this todo in this pull request? It looks pretty straightforward. Is there something undocumented about the complexity of this todo?

Reply:
These can be potentially semver major (especially deprecated(), it also depends on the release line). I am not very keen on looking into the breakage right now.